pkg/daemon/ceph/client/pool.go

```diff
@@ -183,11 +183,24 @@ func CreatePool(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec
 }
 
 func CreatePoolWithPGs(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, pool cephv1.NamedPoolSpec, appName, pgCount string) error {
+	autoscaleMode := pool.AutoscaleMode
 	if pool.Name == "" {
 		return errors.New("pool name must be specified")
 	}
+	if pool.Parameters != nil {
+		placementGroupCount, ok := pool.Parameters["pg_num"]
+		if ok {
+			pgCount = placementGroupCount
+			if autoscaleMode == cephv1.PoolAutoscaleModeOn {
+				logger.Warning("The PGs will be managed by the Ceph PG autoscaler, because of the autoscale mode is on, the settings of the PGs is invalid")
```

Comment:
This seems like a good check to me. I appreciate the thoroughness. If the user is setting pg_num and `autoscaleMode` at the same time, that is not a valid configuration Rook can support. Rook will constantly check to see that the pg_num is set, whereas Ceph will merely treat it as an initial condition. However, Rook should return an error in this case, and not a warning. Rook cannot know which setting the user would prefer to set, and it must not create or modify the pool based on an invalid assumption. By returning an error, it ensures the user corrects the mistake to their own needs.

Reply:
@BlaineEXE yes， what's your suggestions is？ @travisn